lib/events/events.go

```diff
@@ -409,12 +422,18 @@ func (s *Subscription) Poll(timeout time.Duration) (Event, error) {
 	}
 }
 
-func (s *Subscription) C() <-chan Event {
+func (s *subscription) C() <-chan Event {
 	return s.events
 }
 
+func (s *subscription) Unsubscribe() {
+	s.toUnsubscribe <- s
+	for range s.events {
```

Comment:
This is probably fine, yet it tickles my spider sense for possible deadlocks given events are processed by the same loop that handles the unsubscribe and the unsubscribe channels is unbuffered...

Reply:
I was under the incorrect assumption that previously `Unsubscribe` blocked until the logger was actually done unsubscribing and wanted to not change behaviour. So for the same reason (and your spider sense) I am removing it again - thanks.